tests/functional/container/test_lifecycle.py

```diff
@@ -402,3 +402,59 @@ def test_execute_expiration_on_missing_objects(self):
         self.assertEqual(4, len(results))
         for res in results:
             self.assertIsInstance(res[3], Exception)
+
+    def test_execute_exceeding_version_expiration_without_versioning(self):
+        source = """
+        <LifecycleConfiguration>
+            <Rule>
+                <ID>rule1</ID>
+                <Filter>
+                </Filter>
+                <Status>Enabled</Status>
+                <ExceedingVersionExpiration>
+                </ExceedingVersionExpiration>
```

Comment:
To be consistent, we may need to put that inside `<NonCurrentVersionExpiration>`, and make it configurable (independently from the versioning configuration of the container).